Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM docker.io/library/rust:1.69-bookworm AS builder
+
+ARG WASM_SNIP_VERSION_SPEC="^0.4"
+ARG WASM_GC_VERSION_SPEC="^0.1"
+ARG WASM_OPT_VERSION_SPEC="^0.112"
+
+RUN rustup target install wasm32-unknown-unknown \
+ && cargo install --version "${WASM_SNIP_VERSION_SPEC}" wasm-snip \
+ && cargo install --version "${WASM_GC_VERSION_SPEC}" wasm-gc \
+ && cargo install --version "${WASM_OPT_VERSION_SPEC}" wasm-opt \
+ && cargo new --lib /fetcher
+
+COPY Cargo.toml Cargo.lock /
+
+RUN mv /Cargo.* /fetcher \
+ && cd /fetcher \
+ && cargo fetch \
+ && cd / \
+ && rm -rf /fetcher
+
+COPY . /build/
+
+RUN cd build/ && \
+    make release
+
+# Real build
+FROM scratch
+     
+LABEL type=application/vnd.module.wasm.content.layer.v1+wasm 
+COPY --from=builder /build/container/manifest.yaml /build/container/plugin.wasm ./
```

Comment:
`./` should mean this directory. Not sure if `WORKDIR` would work since we are referencing different step

Reply:
well, that `./` makes reference to the current stage and image. What is the absolute path of `./`? having a `WORKDIR` it would be explicit. But maybe the location should be one expected by wasm engine, hence `/manifest.yaml` and `/plugin.wasm`

Reply:
The point of that command is to copy both manifest and plugin from some directory in other image to a root directory of this image. Should I add WORKDIR to the builder image or this image?

Reply:
Then, WORKDIR is not required. It should be:

```
COPY --from=builder /build/container/manifest.yaml /manifest.yaml
COPY --from=builder /build/container/plugin.wasm /plugin.wasm
``

Reply:
really? I do not get it. Anyway, `COPY --from=builder /build/container/manifest.yaml /build/container/plugin.wasm /` seems correct as well to me.